processor/src/stack/overflow.rs

```diff
@@ -0,0 +1,94 @@
+use super::Felt;
+use winter_utils::collections::BTreeMap;
+
+// OVERFLOW
+// ================================================================================================
+
+/// Overflow table stores a list of overflow values from the stack.
+/// We create a separate structure to be able to store historical changes
+/// in the overflow table, so we are able to reconstruct what's in the overflow table
+/// at any given point in time.
+#[derive(Clone)]
+pub struct OverflowTable {
+    active: Vec<Felt>,
+    trace: BTreeMap<usize, Vec<Felt>>,
+    trace_enabled: bool,
+}
+
+impl OverflowTable {
+    pub fn new(trace_enabled: bool) -> Self {
+        Self {
+            active: Vec::new(),
+            trace: BTreeMap::new(),
+            trace_enabled,
+        }
+    }
+
+    pub fn push(&mut self, step: usize, value: Felt) {
+        self.active.push(value);
+        if self.trace_enabled {
+            self.trace.insert(step, self.active.clone());
+        }
+    }
+
+    pub fn pop(&mut self, step: usize) -> Option<Felt> {
+        let popped = self.active.pop();
+        if popped.is_some() && self.trace_enabled {
+            self.trace.insert(step, self.active.clone());
+        }
+
+        popped
+    }
+
+    /// Add values up to a certain step from the overflow table.
+    pub fn append_state_into(&self, step: usize, values: &mut Vec<Felt>) {
+        if let Some(x) = self.trace.range(0..=step).last() {
+            for item in x.1.iter().rev() {
+                values.push(*item);
+            }
+        }
+    }
+
+    /// Add specified size of values from the overflow table into a target vector.
+    pub fn append_into(&self, size: usize, vec: &mut Vec<Felt>) {
+        vec.extend_from_slice(&self.active.as_slice()[..size]);
```

Comment:
Do we need to reverse the slice here as well?

Reply:
I don't think so, we didn't use to reverse it before so I don't tihnk we need to change that logic